bats_ai/core/views/recording.py

```diff
@@ -193,6 +205,10 @@ def update_recording(request: HttpRequest, id: int, recording_data: RecordingUpl
         recording.species_list = recording_data.species_list
     if recording_data.unusual_occurrences:
         recording.unusual_occurrences = recording_data.unusual_occurrences
+    if recording_data.tags:
+        for tag in recording_data.tags:
+            tag, _ = RecordingTag.objects.get_or_create(user=request.user, text=tag)
+        recording.tags.add(tag)
 
```

Comment:
I think your recordint.tags.add needed to be indented in or it was only going to inject the last tag if multiples were submitted

I also think you want some capability to remove any previous tags that aren't found in the updated tags sent by the client.

```suggestion
    if recording_data.tags:
        existing_tags = recording.tags.all()
        for tag in recording_data.tags:
            tag, _ = RecordingTag.objects.get_or_create(user=request.user, text=tag)
            if tag not in existing_tags:
                recording.tags.add(tag)
        # Remove any tags that are not in the updated list
        for existing_tag in existing_tags:
            if existing_tag.text not in recording_data.tags:
                recording.tags.remove(existing_tag)

```